### PR TITLE
12.0 mig product usability

### PR DESCRIPTION
**Pull request by rvalyi**
cc @bealdav @alexis-via 

Note that since v11.0 there is no more product category type field. I believe this is replaced by the new native system to put tags or multiple categories on products. So I removed the related code. The rest of the migration is pretty trivial.

### CODE REVIEW

File: product_usability/pricelist.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+from odoo import models, fields
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    company_id = fields.Many2one(
+        default=lambda self: self.env['res.company']._company_default_get(
+            'product.pricelist'))
```
Comment (bealdav) [inline product_usability/pricelist.py]: useless agrument here
https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_company.py#L164-L169

File: product_usability/product.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            'domain': "[('id', 'in', %s)]" % products.ids,
+        })
+        return action
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    default_code = fields.Char(track_visibility='onchange', copy=False)
+    barcode = fields.Char(track_visibility='onchange')
```
Comment (bealdav) [inline product_usability/product.py]: maybe copy=False here to

**Review by bealdav — APPROVED** [review]
non blocking remarks

**Comment by rvalyi** [comment]
thanks for the review @bealdav. I think we can now merge.

**Pull request merged**